src/container/cli.c

```diff
@@ -487,13 +487,93 @@ struct cont_open_args {
 	daos_handle_t		*hdlp;
 };
 
+struct pmap_refresh_cb_arg {
+	struct dc_pool		*pra_pool;
+	uint32_t		 pra_pm_ver;
+};
+
+static int
+pmap_refresh_cb(tse_task_t *task, void *data)
+{
+	struct pmap_refresh_cb_arg	*cb_arg;
+	struct dc_pool			*pool;
+	uint32_t			 pm_ver;
+	int				 rc = task->dt_result;
+
+	cb_arg = (struct pmap_refresh_cb_arg *)data;
+	pool = cb_arg->pra_pool;
+	if (rc)
+		goto out;
+
+	pm_ver = pool_map_get_version(pool->dp_map);
+	if (pm_ver < cb_arg->pra_pm_ver) {
+		/* very corner case, ask user to try-again-later */
+		rc = -DER_AGAIN;
```

Comment:
i think this might be problematic. users don't handle DER_AGAIN and it would be better IMO if we retry internally.

Reply:
hmm, probably return ESTALE, so client can retry and wait until the server refresh it s pool map?

Reply:
TBH, I was hesitate on this as well. Now dc_pool_query() cannot handle retry, if we want to retry here in DAOS, need to create another new task and retry that task, and even we retry it here in theory it possibly need to retry multiple times, although it is very corner case. So I return EAGAIN to simplify it, I'll dig it a little bit more later.

BTW, ESTALE also no code can automatically handle it, no? @wangdi1

Reply:
added internal retry.